rudder-core/src/main/scala/com/normation/rudder/services/reports/ExecutionBatch.scala

```diff
@@ -588,22 +623,26 @@ object ExecutionBatch extends Loggable {
           , NoAnswerReportType
         )
 
-      case UnexpectedVersion(runTime, runVersion, runExpiration, expectedVersion, expectedExpiration) =>
+      case UnexpectedVersion(runTime, Some(runVersion), runExpiration, expectedVersion, expectedExpiration) =>
```

Comment:
We have this definition: UnexpectedVersion(
    lastRunDateTime   : DateTime
  , lastRunConfigInfo : Some[NodeConfigIdInfo]
  , lastRunExpiration : DateTime
  , expectedConfigInfo: NodeConfigIdInfo
  , expectedExpiration: DateTime
)

So we can't have anything but a some for unexpected version.
It was the path of least modification. An other solution would have been to rework the whole inheritance hierarchy to change what is LastRunAvailable, but it would have been much more work (it was attempted before reverting to that solution).